src/SharpCompress/Utility.cs

```diff
@@ -277,8 +277,38 @@ public static long TransferTo(this Stream source, Stream destination)
             long total = 0;
             while (ReadTransferBlock(source, array, out var count))
             {
+                destination.Write(array, 0, count);
                 total += count;
+            }
+            return total;
+        }
+        finally
+        {
+            ArrayPool<byte>.Shared.Return(array);
+        }
+    }
+
+    public static long TransferTo(this Stream source, Stream destination, long maxLength)
+    {
+        var array = GetTransferByteArray();
+        try
+        {
+            long total = 0;
+            var remaining = maxLength;
+            var maxReadSize = array.Length;
+            if (remaining < maxLength)
```

Comment:
This conditions is impossible because remaining = maxLength
It should be if (remaining < maxReadSize )

Reply:
you're right